builtin/json.mbt

```diff
@@ -124,8 +124,8 @@ pub fn Map::to_json[K : Show, V : ToJson](self : Map[K, V]) -> Json {
 
 pub fn Option::to_json[T : ToJson](self : T?) -> Json {
   match self {
-    None => Null
-    Some(value) => value.to_json()
```

Comment:
if we follow the json format generated by enum type, the mapping between Option[T] and json should be

```
None   <->  { "$tag": "None" }
Some(x)  <->  { "$tag": "Some", 0: x }
```

It seems that using a special json format like this saves more memory. It looks good to me, but I am not sure what problems it might cause...

Anyway, the `"None"` can be omitted here:

```
None       <->   nulll
Some(x)  <->  { "Some": x }
```